hooks/boot/05-data-helpers.rb

```diff
@@ -0,0 +1,135 @@
+  end
+
+  def pg_sql_statement(config, statement)
+    pg_command_base(config, 'psql', "-d #{config[:database]} -t -c \"" + statement + '"')
+  end
+
+  # WARNING: deletes all the data from a database. No warnings. No confirmations.
+  def empty_database!(config)
+    generate_delete_statements = pg_sql_statement(config, %q(
+          select string_agg('drop table if exists \"' || tablename || '\" cascade;', '')
+          from pg_tables
+          where schemaname = 'public';
+        ))
+    delete_statements = `#{generate_delete_statements}`
+    execute(pg_sql_statement(config, delete_statements)) if delete_statements
+  end
+
+  def empty_db_in_postgresql(db)
+    logger.info "Dropping #{db} database!"
+
+    config = load_db_config(db)
+    if remote_host?(config[:host])
+      empty_database!(config)
+    else
+      execute("sudo -u postgres dropdb #{config[:database]}")
+    end
+  end
+
+  def reset_candlepin
+    execute('rm -f /var/lib/candlepin/{cpdb_done,cpinit_done}')
+    empty_db_in_postgresql(candlepin)
+  end
+
+  def load_mongo_config
+    seeds = param_value('katello', 'pulp_db_seeds')
+    seed = seeds.split(',').first
+    host, port = seed.split(':') if seed
+    {
+      host: host || 'localhost',
+      port: port || '27017',
+      database: param_value('katello', 'pulp_db_name') || 'pulp_database',
+      username: param_value('katello', 'pulp_db_username'),
+      password: param_value('katello', 'pulp_db_password'),
+      ssl: param_value('katello', 'pulp_db_ssl') || false,
+      ca_path: param_value('katello', 'pulp_db_ca_path'),
+      ssl_certfile: param_value('katello', 'pulp_db_ssl_certfile'),
+    }
+  end
+
+  def empty_mongo
+    logger.info "Dropping Pulp database!"
+
+    mongo_config = load_mongo_config
+    if remote_host?(mongo_config[:host])
+      empty_remote_mongo(mongo_config)
+    else
+      start_services(['rh-mongodb34-mongod'])
+      execute("mongo #{mongo_config[:database]} --eval 'db.dropDatabase();'")
+    end
+  end
+
+  def empty_remote_mongo(config)
+    if config[:ssl]
+      ssl = "--ssl"
+      if config[:ca_path]
+        ca_cert = "--sslCAFile #{config[:ca_path]}"
+        client_cert = "--sslPEMKeyFile #{config[:ssl_certfile]}" if config[:ssl_certfile]
+      end
+    end
+    username = "-u #{config[:username]}" if config[:username]
+    password = "-p #{config[:password]}" if config[:password]
+    host = "--host #{config[:host]} --port #{config[:port]}"
+    cmd = "mongo #{config[:database]} #{username} #{password} #{host} #{ssl} #{ca_cert} #{client_cert} --eval 'db.dropDatabase();'"
+    execute(cmd)
+  end
+
+  def reset_pulp
+    execute('rm -f /var/lib/pulp/init.flag')
+    empty_mongo
+    execute('rm -rf /var/lib/pulp/{distributions,published,repos}/*')
+  end
+end
+
+def clear_pulpcore_content
+  if File.directory?('/var/lib/pulp/docroot')
+    FileUtils.rm_rf(Dir.glob("/var/lib/pulp/docroot/*"))
```

Comment:
The installer can recreate the docroot. Can't we just remove `/var/lib/pulp/docroot` instead of globbing?

Reply:
I'll change this to `execute('rm -rf /var/lib/pulp/docroot/*')` to be consistent with what we are already doing in other hooks and helpers

Reply:
Well, that still ends up globbing. I also think it doesn't remove hidden files. I think the previous approach was taken because `/var/lib/pulp` may be a mountpoint and you can't remove that. However, the media storage shouldn't be on its own mountpoint (all the temp dirs should be on the same mount). That means I think it's safe to remove the whole directory and let the installer recreate it.